packages/react-instantsearch-core/src/hooks/useConnector.ts

```diff
@@ -116,7 +117,12 @@ export function useConnector<
         helper,
         parent: parentIndex,
         instantSearchInstance: search,
-        results,
+        results:
+          widget.dependsOn === 'recommend'
+            ? // @TODO: this is to avoid using wrong hits in SSR,
+              // will be replace with SSR support for recommend
+              (null as unknown as algoliasearchHelper.SearchResults<any>)
```

Comment:
as it's a type, you can do `import type { SearchResults }` but it doesn't really change anything